### DISCUSSION TITLE
Ant Design 弹窗封装：易用性和可维护性 | whincwu(右耳朵猫)

### DISCUSSION THREAD

**Discussion by giscus** [whinc/blog#37] [Announcements]
# Ant Design 弹窗封装：易用性和可维护性 | whincwu(右耳朵猫)

在 Ant Design(简称 Antd) 组件库中弹窗的使用频率很高，Antd 提供的Modal组件一般用法如下：

https://whinc.github.io/blog/2023/03/04/antd-create-modal/

**Comment by qingshan-15** [comment]
方案三为什么会报错？函数式组件不能用静态方法嘛？创建命名空间的方式并不能完整解决啊。另外就是一些类型检查是过不去的，比如传入的ContentConponent，_payload

**Reply by whinc** [reply]
参考上面demo（可以在线运行预览），ts类型方面 nice-modal-react 还不够好，可以暂时通过 props as type 方式显式指明类型
